File: geoquake/src/main/java/com/geo/geoquake/models/Prefs.java
package com.geo.geoquake.models;

import android.content.Context;
import android.content.SharedPreferences;

import com.geo.geoquake.App;
import com.geo.geoquake.GeoQuakeDB;

import java.util.Objects;

/**
 * Created by George Stinson on 2017-03-28.
 */

public class Prefs {

    public static final String QUAKE_PREFS = "quake_prefs";

    private static Prefs prefs;
    private final SharedPreferences sharedPreferences;

    public static Prefs getInstance() {
        if(prefs == null) {
            prefs = new Prefs();
        }
        return prefs;
    }

    private Prefs() {
        this.sharedPreferences = Objects.requireNonNull(App.Companion.getAppContext())
                .getSharedPreferences(QUAKE_PREFS, Context.MODE_PRIVATE);
    }

    public void setRefreshLimiter() {
        SharedPreferences.Editor e = sharedPreferences.edit();
        e.putLong(KEYS.REFRESH_LIMITER, GeoQuakeDB.getTime());
        e.apply();
    }

    public long getRefreshLimiter() {
        return sharedPreferences.getLong(KEYS.REFRESH_LIMITER, 0);
    }

    public void setCacheTime(String cacheTime) {
        SharedPreferences.Editor e = sharedPreferences.edit();
        e.putString(KEYS.CACHE_TIME, cacheTime);
        e.apply();
    }

    public String getCacheTime() {
        return sharedPreferences.getString(KEYS.CACHE_TIME, "300000");
    }

    public void setSource(int source) {
        SharedPreferences.Editor e = sharedPreferences.edit();
        e.putInt(KEYS.SOURCE, source);
        e.apply();
    }

    public int getSource() {
        return sharedPreferences.getInt(KEYS.SOURCE, 0);
    }

    private static class KEYS {
        private static final String REFRESH_LIMITER = "refresh_limiter";
        private static final String CACHE_TIME = "cache_time";
        private static final String SOURCE = "source";
    }
}

File: geoquake/src/main/java/com/geo/geoquake/models/Prefs.kt
package com.geo.geoquake.models

import android.content.Context
import android.content.SharedPreferences
import com.geo.geoquake.App.Companion.appContext
import com.geo.geoquake.GeoQuakeDB
import java.util.*

/**
 * Created by George Stinson on 2017-03-28.
 */
class Prefs private constructor() {
    private val sharedPreferences: SharedPreferences
    fun setRefreshLimiter() {
        val e = sharedPreferences.edit()
        e.putLong(KEYS.REFRESH_LIMITER, GeoQuakeDB.getTime())
        e.apply()
    }

    val refreshLimiter: Long
        get() = sharedPreferences.getLong(KEYS.REFRESH_LIMITER, 0)

    var cacheTime: String?
        get() = sharedPreferences.getString(KEYS.CACHE_TIME, "300000")
        set(cacheTime) {
            val e = sharedPreferences.edit()
            e.putString(KEYS.CACHE_TIME, cacheTime)
            e.apply()
        }

    var source: Int
        get() = sharedPreferences.getInt(KEYS.SOURCE, 0)
        set(source) {
            val e = sharedPreferences.edit()
            e.putInt(KEYS.SOURCE, source)
            e.apply()
        }

    private object KEYS {
        const val REFRESH_LIMITER = "refresh_limiter"
        const val CACHE_TIME = "cache_time"
        const val SOURCE = "source"
    }

    companion object {
        const val QUAKE_PREFS = "quake_prefs"
        private var prefs: Prefs? = null
        @JvmStatic
        val instance: Prefs?
            get() {
                if (prefs == null) {
                    prefs = Prefs()
                }
                return prefs
            }
    }

    init {
        sharedPreferences = appContext!!.getSharedPreferences(QUAKE_PREFS, Context.MODE_PRIVATE)
    }
}